### PR TITLE
[WIP] Add practice exercise: alphametics

### PR DESCRIPTION
**Pull request by SaschaMann**
I'm struggling to come up with a good enough solution that passes tests in time without relying on packages. If you want to give it a go, help would be appreciated :)

### CODE REVIEW

File: exercises/practice/alphametics/runtests.jl [hunk 1]
```diff
@@ -0,0 +1,102 @@
+using Test
+
+include("alphametics.jl")
+
+@testset "puzzle with three letters" begin
+    @test solve("I + BB == ILL") == Dict(
+        'I' => 1,
+        'B' => 9,
+        'L' => 0
```
Comment (SaschaMann) [inline exercises/practice/alphametics/runtests.jl:9]: Might be better to change these to `Symbol`s but I'm not sure

Comment (cmcaine) [reply exercises/practice/alphametics/runtests.jl:9]: Chars are better here, imo

**Comment by cmcaine** [comment]
Here's a brute-force solution that's probably fast enough to pass the CI.

https://gist.github.com/cmcaine/2a056d6b8926e0ae033fe041c5d5aec7

I should probably have timed myself, but to give an idea of the difficulty, I read the prompt on Thursday evening (2 days ago) and thought about the problem for probably 1-3 hours while walking the dog, etc. Then I wrote this solution over about 2-2.5h today.

The solution uses some fairly advanced Julia knowledge: metaprogramming (but you could do string manipulation and Meta.parse to simplify it a bit) and `Base.invokelatest` with a function barrier.

**Comment by cmcaine** [comment]
Oh, I used `Combinatorics.jl` for `permutations`, but that bit could be replaced fairly straightforwardly.

**Comment by cmcaine** [comment]
The "Efficient Brute Force" approach described in the British informatics olympiad solutions would probably perform well in Julia, too.

That doesn't require any metaprogramming. Instead you represent the problem as one or two matrices. SEND MORE MONEY would look like:

```
#    S     E     N     D     M     O     R     Y
A = [1000  100   10    1     0     0     0     0; # SEND
     0     1     0     0     1000  100   10    0] # MORE

B = [0     10    100   0     10000 1000  0     1] # MONEY

# Solution
X = [9 5 6 7 1 0 8 2]

is_valid(A, B, X) == sum(A .* X) == sum(B .* X)
```

**Comment by SaschaMann** [comment]
> Oh, I used Combinatorics.jl for permutations, but that bit could be replaced fairly straightforwardly.

That's not an issue, the definition is short enough we can just copy it and add the copyright notice in the file itself.

**Pull request merged**